### PR TITLE
Add option to exclude groups from creating link shares

### PR DESCRIPTION
**Pull request by icewind1991**
I would have hoped to solve this in a more generic "per group share settings" way but the way the code base interacts with the share settings is to messy to do that without a lot of rework (this PR includes *some* improvement in this area).

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
How is this then shown to clients?
Is then just files_sharing -> public -> enabled: false?
(in capabilities)
So nothing needs to be done/tested on clients?

**Comment by icewind1991** [comment]
Clients should see the same from the capabilities response as if link shares were completely disabled

**Review by rullzer — APPROVED** [review]
Does the trick for me!

**Comment by rullzer** [comment]
However, linting, psalm and phpunit are not happy

File: apps/files/list.php [hunk 1]
```diff
@@ -22,10 +22,14 @@
...
 
+use OCP\Share\IManager;
+
 $config = \OC::$server->getConfig();
 $userSession = \OC::$server->getUserSession();
 // TODO: move this to the generated config.js
-$publicUploadEnabled = $config->getAppValue('core', 'shareapi_allow_public_upload', 'yes');
+/** @var IManager $shareManager */
+$shareManager = \OC::$server->get(IManager::class);
+$publicUploadEnabled = $shareManager->shareApiLinkAllowPublicUpload();
```
Comment (juliushaertl) [inline apps/files/list.php:32]: I couldn't find any usage of the data attribute where this value is assigned to but from a compatibility perspective we should probably preserve the yes/no values here:

```suggestion
$publicUploadEnabled = $shareManager->shareApiLinkAllowPublicUpload() ? 'yes' : 'no';
```

**Review by juliusknorr — CHANGES_REQUESTED** [review]
Small comment, otherwise code looks good and works as expected :+1:

**Review by juliusknorr — APPROVED** [review]

**Comment by MorrisJobke** [comment]
CI is happy :)

**Pull request merged**